services/skus/radom/event_test.go

```diff
@@ -0,0 +1,985 @@
+package radom
+
+import (
+	"context"
+	"testing"
+
+	uuid "github.com/satori/go.uuid"
+	should "github.com/stretchr/testify/assert"
+	must "github.com/stretchr/testify/require"
+)
+
+func TestParseEvent(t *testing.T) {
+	type tcGiven struct {
+		rawEvent string
+	}
+
+	type tcExpected struct {
+		event   *Event
+		mustErr must.ErrorAssertionFunc
+	}
+
+	type testCase struct {
+		name  string
+		given tcGiven
+		exp   tcExpected
+	}
+
+	tests := []testCase{
+		{
+			name: "new_subscription",
+			given: tcGiven{
+				rawEvent: `{
+				  "eventType": "newSubscription",
+				  "eventData": {
+					"newSubscription": {
+					  "subscriptionId": "54453f86-8cfa-4eee-8818-050fc61f560b"
+					}
+				  },
+				  "radomData": {
+					"checkoutSession": {
+					  "checkoutSessionId": "71da4f76-0ac7-47ee-bb51-9d9577232245",
+					  "metadata": [
+						{
+						  "key": "brave_order_id",
+						  "value": "b5269191-1d8d-4934-b105-3221da010222"
+						}
+					  ]
+					}
+				  }
+				}`,
+			},
+			exp: tcExpected{
+				event: &Event{
+					EventType: "newSubscription",
+					EventData: &EventData{
+						New: &NewSubscription{
+							SubscriptionID: uuid.FromStringOrNil("54453f86-8cfa-4eee-8818-050fc61f560b"),
```

Comment:
The input is not dynamic so why would it fail if passed a valid uuid?

Reply:
> Its more convenient and less distracting, plus given the input is not dynamic why would it fail if passed a valid uuid?
I agree, and as alluded in the comment, it's a question of the library of choice, because the counter part used in other places does have `RequireFromString` which is exactly what's neccessary.

In my view, and my concern, is that convenience is not a sufficient argument to use a method that _might_ hide a problem.

> Hopefully the test would fail given invalid input.
I would not hope for that. We don't expect every single piece under testing to verify that the supplied UUID is not an empty one.

And another reason I argue about this is specifically because myself I do not use it even though considered from the same perspective as you have, and decided that correctness was more important. Plus, it's just a few more symbols.

In any case, let this not block us, we can discuss it later.

Reply:
> In my view, and my concern, is that convenience is not a sufficient argument to use a method that might hide a problem.

Its not just convenience though is it... discussing later in more detail sounds like a plan.